commons/src/test/java/io/aiven/kafka/connect/common/source/input/ByteArrayTransformerTest.java

```diff
@@ -69,9 +75,39 @@ void testGetRecordsEmptyInputStream() {
 
         final IOSupplier<InputStream> inputStreamIOSupplier = () -> inputStream;
 
-        final Stream<SchemaAndValue> records = byteArrayTransformer.getRecords(inputStreamIOSupplier, TEST_TOPIC, 0,
+        final Stream<SchemaAndValue> records = byteArrayTransformer.getRecords(inputStreamIOSupplier, 0, TEST_TOPIC, 0,
                 sourceCommonConfig, 0);
 
         assertThat(records).hasSize(0);
     }
+
+    /**
+     * @param maxBufferSize
+     *            the maximum buffer size
+     * @param numberOfExpectedRecords
+     *            the number of records the byte array is split into based off the max buffer size
+     */
+    @ParameterizedTest
+    @CsvSource({ "1,10", "2,5", "3,4", "4,3", "5,2", "6,2", "7,2", "8,2", "9,2", "10,1", "11,1", "12,1" })
+    void testGetRecordsWithVariableMaxBufferSize(final int maxBufferSize, final int numberOfExpectedRecords) {
```

Comment:
Can we add a test where input rec size is beyond 4096, and if user does not configure transformer.max.buffer.size.